src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -466,18 +466,46 @@ private EmailWrapper generateFeedbackSessionEmailBaseForInstructors(
+     * {@code userName} and {@code userEmail}.
+     */
     private EmailWrapper generateFeedbackSessionClosedEmail(CourseAttributes course,
             FeedbackSessionAttributes session, String userName, String userEmail) {
+
         String template = EmailTemplates.USER_FEEDBACK_SESSION_CLOSED;
+
         String subject = EmailType.FEEDBACK_CLOSED.getSubject();
 
-        String emailBody = Templates.populateTemplate(template,
-                "${userName}", SanitizationHelper.sanitizeForHtml(userName),
-                "${courseName}", SanitizationHelper.sanitizeForHtml(course.getName()),
-                "${courseId}", SanitizationHelper.sanitizeForHtml(course.getId()),
-                "${feedbackSessionName}", SanitizationHelper.sanitizeForHtml(session.getFeedbackSessionName()),
-                "${deadline}", SanitizationHelper.sanitizeForHtml(TimeHelper.formatTime12H(session.getEndTime())),
-                "${supportEmail}", Config.SUPPORT_EMAIL);
+        String emailBody;
+
+        /*Check if the given userName and userEmail correspond
+        to an Instructor or a Student. */
+        if (instructorsLogic.isEmailOfInstructorOfCourse(userEmail, userName)) {
+            emailBody = Templates.populateTemplate(template,
+                    "${userName}", SanitizationHelper.sanitizeForHtml(userName),
+                    "${instructorFragment}",
+                            "<p>The email below has been sent to students of course: "
+                            + SanitizationHelper.sanitizeForHtml(course.getId()) + ", "
+                            + SanitizationHelper.sanitizeForHtml(course.getName())
+                            + ".<br>" + Const.EOL + "<br>" + Const.EOL
+                            + "=== Email message as seen by the students ===</p>" + Const.EOL,
+                    "${courseName}", SanitizationHelper.sanitizeForHtml(course.getName()),
+                    "${courseId}", SanitizationHelper.sanitizeForHtml(course.getId()),
+                    "${feedbackSessionName}", SanitizationHelper.sanitizeForHtml(session.getFeedbackSessionName()),
+                    "${deadline}", SanitizationHelper.sanitizeForHtml(TimeHelper.formatTime12H(session.getEndTime())),
+                    "${supportEmail}", Config.SUPPORT_EMAIL);
+        } else {
+            emailBody = Templates.populateTemplate(template,
+                    "${userName}", SanitizationHelper.sanitizeForHtml(userName),
+                    "${instructorFragment}", "",
+                    "${courseName}", SanitizationHelper.sanitizeForHtml(course.getName()),
+                    "${courseId}", SanitizationHelper.sanitizeForHtml(course.getId()),
+                    "${feedbackSessionName}", SanitizationHelper.sanitizeForHtml(session.getFeedbackSessionName()),
+                    "${deadline}", SanitizationHelper.sanitizeForHtml(TimeHelper.formatTime12H(session.getEndTime())),
+                    "${supportEmail}", Config.SUPPORT_EMAIL);
```

Comment:
I do believe that this is the right type of solving this issue. If you have a look at the methods `generateFeedbackSessionEmailBaseForStudents` and `generateFeedbackSessionEmailBaseForInstructors` above (lines 391-451) you will see that there are repetitions as well. What else do you propose me to do? I would prefer to be given some tips in order to improve my solution rather than just a message telling me that I did not do anything. Thank you.